### ISSUE TITLE
Unhandled Exception DLL Not found

### ISSUE THREAD

**Issue by Lucyhutch50** [marco110/LibPostalNet_1#2]
Hi,
I have tried building this code and running the console application.  I have previously downloaded and built the windows DLL  and did the git hub pull and build using Msys64.

I have used the links in the readme file and tried compiling the project in Visual studio.  Each time I attempt to execute the libpostalconsole application it crashes...

C:\LibPostalNet_1-master\LibPostalConsole\bin\Release>libpostalconsole
Loading LibPostal...

Unhandled Exception: System.DllNotFoundException: Unable to load DLL 'libpostal': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
   at LibPostalNet.LibPostal.UnsafeNativeMethods.Setup()
   at LibPostalNet.LibPostal..ctor(String dataDir) in C:\LibPostalNet_1-master\LibPostalNet\LibPostal.cs:line 69
   at LibPostalNet.LibPostal.GetInstance(String dataDir) in C:\LibPostalNet_1-master\LibPostalNet\LibPostal.cs:line 12
   at LibPostalNet.LibPostal.GetInstance() in C:\LibPostalNet_1-master\LibPostalNet\LibPostal.cs:line 11
   at LibPostalConsole.Program.Main(String[] args) in C:\LibPostalNet_1-master\LibPostalConsole\Program.cs:line 13

Can anyone help?
Thanks in advance
Phil

@marco110

**Comment by Lucyhutch50** [comment]
@mapo80 @AeroXuk   Can you help gents?

**Comment by AeroXuk** [comment]
Within your `C:\LibPostalNet_1-master\LibPostalConsole\bin\Release` directory there should be two folder called `x86` and `x64`.  You will need to place the precompiled libpostal.dll into the relevant ones.

[x86 / 32-bit version](https://goo.gl/Bf3EzE) into `x86`
[x64 / 64-bit version](https://goo.gl/o8DAi8) into `x64`

**Comment by Lucyhutch50** [comment]
I have done that and then try to execute the libconsole application and get
that error message,  is it trying to find the data files in a particular
place or method?
Thanks
Phil

**Comment by AeroXuk** [comment]
> I have done that and then try to execute the libconsole application and get that error message, is it trying to find the data files in a particular place or method? Thanks Phil

The precompiled copy of LibPostal for Windows looks in `C:\libpostal\` for the data model.

I think the issue is most likely going to be with something here:
https://github.com/marco110/LibPostalNet_1/blob/9568473c851f2fbaad246faaf3fa06987d7fd612/LibPostalNet/LibPostal.cs#L55-L68
This section should automaticaly run when LibPostalNet is first referenced, it detemins if it's running in 32 or 64 bit mode and adds the directory to the .dll search list.

If you know which version (32 or 64 bit) you will be running you can place the .dll directy within the bin folder.

**Comment by Lucyhutch50** [comment]
@AeroXuk  I tried placing first the 32 bit version into the path and this gives the error I already placed above and if I place the 64 bit version into this area I get a different error.

When I tried debugging the source code within Visual studio it appeared that datadir , datapath had no settings ?

I am trying to get the Libpostal.net to load as I would like to try and use this component or libpostal dll in a delphi program but until I can prove that the base DLL is working as expected I don't see trying to code it into Delphi.

I managed to build a version of libpostal in a WSL machine which is why I would like a native version for  Windows.  I am on windows 10 - does this have any significance and the .NET framework is 4.61

Thanks
Phil

**Comment by AeroXuk** [comment]
> @AeroXuk I tried placing first the 32 bit version into the path and this gives the error I already placed above and if I place the 64 bit version into this area I get a different error.

If you're getting a different error with the 64-bit version that could be that the .dll is now getting loaded. But something else is also not right.  Could you post the other error message?

> When I tried debugging the source code within Visual studio it appeared that datadir , datapath had no settings ?

You can manually set the path to the libpostal data dir on the `GetInstance()` call.

    LibPostal.GetInstance( "C:\Libpostal\" );

**Comment by Lucyhutch50** [comment]
@AeroXuk 

Thanks for the help so far.

I tried editing the path and it seemed to take longer to error but then came up with

Loading LibPostal...

Unhandled Exception: System.DllNotFoundException: Unable to load DLL 'libpostal': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
   at LibPostalNet.LibPostal.UnsafeNativeMethods.SetupDatadir(IntPtr datadir)
   at LibPostalNet.LibPostal..ctor(String dataDir) in C:\LibPostalNet_1-master\LibPostalNet\LibPostal.cs:line 69
   at LibPostalNet.LibPostal.GetInstance(String dataDir) in C:\LibPostalNet_1-master\LibPostalNet\LibPostal.cs:line 12
   at LibPostalConsole.Program.Main(String[] args) in C:\LibPostalNet_1-master\LibPostalConsole\Program.cs:line 13

This was from the a windows CMD window

I then tried replacing the libpostal dll with the 64 bit version and get

Loading LibPostal...

Unhandled Exception: System.BadImageFormatException: An attempt was made to load a program with an incorrect format. (Exception from HRESULT: 0x8007000B)
   at LibPostalNet.LibPostal.UnsafeNativeMethods.SetupDatadir(IntPtr datadir)
   at LibPostalNet.LibPostal..ctor(String dataDir) in C:\LibPostalNet_1-master\LibPostalNet\LibPostal.cs:line 69
   at LibPostalNet.LibPostal.GetInstance(String dataDir) in C:\LibPostalNet_1-master\LibPostalNet\LibPostal.cs:line 12
   at LibPostalConsole.Program.Main(String[] args) in C:\LibPostalNet_1-master\LibPostalConsole\Program.cs:line 13

If you run the libpostal.dll throug the dependancy checker it shows lots of missing DLL's it needs? Is this the issue?

Sorry for all the questions.
Phil
